src/index.ts

```diff
@@ -90,11 +91,27 @@ export class JolocomWebServiceClient {
 
     ws.onmessage = (evt) => {
       console.log('received websocket data', evt.data)
-      let msg
+      let msg: any
       try {
         msg = JSON.parse(evt.data)
         this.messages[msg.id].resolve(msg.response)
         this._finalizeMessage(msg.id)
+
+        if(msg.response && msg.response.id) {
+          if (this.rpcWS)
+              new Promise((resolve, reject) => {
```

Comment:
What does this do?
it seems the value is discarded, and so the resolve on line 110 would basically have no effect (i.e. listening for `Message` resolution would be broken)

Reply:
Many thanks @mnzaki, 
And sorry, this was not all good. Now I fixed it and allowed for a continues receive of follow-up messages.

You can try the flow now with https://github.com/jolocom/interactions-demo/pull/34